src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/CrispImageCache.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.InheritanceMargin.MarginGlyph
+{
+    internal static class CrispImageCache
+    {
+        private static ImageAttributes? s_attributes;
+        private static readonly Dictionary<ImageMoniker, BitmapSource> s_ImageCache = new();
```

Comment:
this all feels very non-threadsafe.

Reply:
Sure, I could add a lock to the write part of this dictionaryary